railgun/internal/manager/controllerdef.go

```diff
@@ -199,49 +199,60 @@ func setupControllers(logger logr.Logger, mgr manager.Manager, proxy proxy.Proxy
 		Disabling Ingress controller`)
 	}
 
-	kongClusterPluginGVR := schema.GroupVersionResource{
-		Group:    konghqcomv1.SchemeGroupVersion.Group,
-		Version:  konghqcomv1.SchemeGroupVersion.Version,
-		Resource: "kongclusterplugins",
-	}
-	if ctrlutils.CRDExists(mgr.GetClient(), kongClusterPluginGVR) {
-		logger.Info("kongclusterplugins.configuration.konghq.com v1beta1 CRD available on cluster.")
-		controller := ControllerDef{
-			IsEnabled: &c.KongClusterPluginEnabled,
-			Controller: &configuration.KongV1KongClusterPluginReconciler{
-				Client:           mgr.GetClient(),
-				Log:              ctrl.Log.WithName("controllers").WithName("KongClusterPlugin"),
-				Scheme:           mgr.GetScheme(),
-				Proxy:            proxy,
-				IngressClassName: c.IngressClassName,
-			},
+	if c.KongClusterPluginEnabled == util.EnablementStatusEnabled {
+		kongClusterPluginGVR := schema.GroupVersionResource{
+			Group:    konghqcomv1.SchemeGroupVersion.Group,
+			Version:  konghqcomv1.SchemeGroupVersion.Version,
+			Resource: "kongclusterplugins",
+		}
+
+		if ctrlutils.CRDExists(mgr.GetClient(), kongClusterPluginGVR) {
+			logger.Info("kongclusterplugins.configuration.konghq.com v1beta1 CRD available on cluster.")
+			controller := ControllerDef{
+				IsEnabled: &c.KongClusterPluginEnabled,
+				Controller: &configuration.KongV1KongClusterPluginReconciler{
+					Client:           mgr.GetClient(),
+					Log:              ctrl.Log.WithName("controllers").WithName("KongClusterPlugin"),
+					Scheme:           mgr.GetScheme(),
+					Proxy:            proxy,
+					IngressClassName: c.IngressClassName,
+				},
+			}
+			controllers = append(controllers, controller)
+		} else {
+			logger.Info(`kongclusterplugins.configuration.konghq.com v1beta1 CRD not available on cluster.
+		Disabling KongClusterPlugin controller`)
 		}
-		controllers = append(controllers, controller)
 	} else {
-		logger.Info(`kongclusterplugins.configuration.konghq.com v1beta1 CRD not available on cluster.
+		logger.Info(`kong cluster plugin is disabled.
 		Disabling KongClusterPlugin controller`)
 	}
+	if c.KnativeIngressEnabled == util.EnablementStatusEnabled {
```

Comment:
Here's where the "auto" option gets a bit tricky: should we treat "auto" differently for a 3rd party CRD like knative?

I think for now we need to do something with it for consistency:

```suggestion
	if c.KnativeIngressEnabled == util.EnablementStatusEnabled || c.KnativeIngressEnabled == util.EnablementStatusAuto {
```

Reply:
Same as above response.